hubblestack/extmods/modules/nebula_osquery.py

```diff
@@ -74,11 +74,32 @@ def queries(query_group,
         salt '*' nebula.queries hour verbose=True
         salt '*' nebula.queries hour pillar_key=sec_osqueries
     '''
+    config = {}
+    if isinstance(query_file, list):
+        for file in query_file:
+            if 'salt://' in file:
+                file = __salt__['cp.cache_file'](file)
+            if os.path.isfile(file):
+                with open(file, 'r') as f:
+                    config = _dict_update(config,
+                                         yaml.safe_load(f),
+                                         recursive_update=True,
+                                         merge_lists=True)
+        if salt.utils.is_windows():
+            query_file = "salt://hubblestack_nebula/hubblestack_nebula_win_queries.yaml"
+        else:
+            query_file = 'salt://hubblestack_nebula/hubblestack_nebula_queries.yaml'
+
+        query_file = __salt__['cp.cache_file'](query_file)
+        with open(query_file,'w') as yaml_file:
+            yaml.dump(config, yaml_file, default_flow_style=False)
```

Comment:
There are a couple of problems here. I can sort of see what you were going for -- let's write it out to a file the logic below doesn't have to be changed. However, it won't work, primarily because the logic below also calls `cp.cache_file` (on line 143/164), which would overwrite your `yaml.dump`ed changes.

It's also not necessary. If we move this list handling logic further below, we can do it all in memory and never have to write to disk.

Here's the basic logic, in pseudocode:

```
if query_file is None:
    <set it to hubblestack_nebula_queries or hubblestack_nebula_win_queries depending on windows/linux>
if query_file is not a list:
    query_file = [query_file]  # This way we can treat all runs the same. Normal runs will just be a list of one file
config = {}
for f in query_file:
    <merge each file into config>
```

At the end of this logic we'll have a `config` variable that is a combination of one or more query files, and we can do the normal osquery run on that data, without ever having to write to disk.